### DISCUSSION TITLE
Upload during entity creation ignores PATH

### DISCUSSION THREAD

**Discussion by Mehrdad-V** [strapi/strapi#7320] [Legacy Topics]
I tested path in upload url of strapi and it works fine but when uploading file during entity creation it ignores path.
I use Amazon S3 provider now and wanted to collect each users' files in his own folder with one api call not one for entry and another for file
what I did is overriding controllers and call upload service in it. do you have any idea or clue about this?

**Comment by derrickmehaffy** [comment]
Can you provide some more examples as to what you tried (maybe a quick sample project you were using) so we can try to reproduce. I just tried it myself with AWS-S3 and Postman and the path variable was working fine in my test.

**Reply by Mehrdad-V** [reply]
In normal upload on /upload endpoint it's OK but if you make an entity with an attachment field, you can't specify path parameter to manage it in AWS-S3 while using upload during entity creation. path is ignored

**Reply by derrickmehaffy** [reply]
Ah yeah I apologize the path parameter isn't supported during entry creation only on a straight upload (in this case you have to create the entry first, then upload the file using the ref information)

**Comment by Mehrdad-V** [comment]
yes, exactly
I customized entity create controller (of curse made a general function) and passed "ctx" to it, and in case of multipart request (with files) the controller will call upload service internally. I make the linking data manually in this controller.

**Reply by derrickmehaffy** [reply]
Did this solve your issue?

**Reply by Mehrdad-V** [reply]
yes, I managed files this way now, but hope there would be a general approach to it in future

**Reply by derrickmehaffy** [reply]
Thank you @Mehrdad-V we will keep that in mind :)
